hibernate-core/src/test/java/org/hibernate/orm/test/id/uuid/sqlrep/sqlchar/UUIDCharTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.id.uuid.sqlrep.sqlchar;
+
+import java.util.UUID;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+
+import org.hibernate.cfg.Configuration;
+import org.hibernate.type.UUIDCharType;
+
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Steve Ebersole
+ */
+public class UUIDCharTest extends BaseCoreFunctionalTestCase {
+
+	private static class UUIDPair {
+		UUID rootId;
+		UUID childId;
+
+		public UUIDPair(UUID rootId, UUID childId) {
+			this.rootId = rootId;
+			this.childId = childId;
+		}
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { Node.class };
+	}
+
+	@Override
+	public void configure(Configuration cfg) {
+		cfg.registerTypeOverride(
+				new UUIDCharType() {
+					@Override
+					protected boolean registerUnderJavaType() {
+						return true;
+					}
+				}
+		);
+	}
+
+	@Test
+	public void testUsage() {
+		final UUIDPair uuidPair = fromTransaction( session -> {
+			final Node root = new Node( "root" );
+			session.save( root );
+			assertThat( root.id, notNullValue());
+
+			final Node child = new Node( "child", root );
+			session.save( child );
+			assertThat( child.id, notNullValue() );
+
+			return new UUIDPair( root.id, child.id );
+		} );
+
+		inTransaction( session -> {
+			final Node root = session.get( Node.class, uuidPair.rootId );
+			assertThat( root, notNullValue() );
+			final Node child = session.get( Node.class, uuidPair.childId );
+			assertThat( child, notNullValue() );
+		} );
+
+		inTransaction( session -> {
+			final Node node = session.createQuery( "from Node n join fetch n.parent where n.parent is not null", Node.class ).uniqueResult();
+			assertThat( node, notNullValue() );
+			assertThat( node.parent, notNullValue() );
+		} );
+	}
+
+	@Entity(name = "Node")
+	static class Node {
+
+		@Id
+		@Column(name = "id", length = 36)
+		@GeneratedValue( generator = "uuid2" )
```

Comment:
Shouldn't this be the following? Either this, or change the usage in the other test to `@GeneratedValue( generator = "uuid2" )` as well.

```suggestion
		@GeneratedValue(generator = "uuid")
		@GenericGenerator(name = "uuid", strategy = "uuid2")
```

Reply:
I changed as you suggested

Reply:
By closer debugging, I found the original testing case is buggy. The gist of testing here is to ensure the uuid db column's db type is binary and (var)char. Previously we used `@Column(length = 16)` and `@Column(length=36)` to achieve the goal, but they both end up with 'binary' though with different lengths.
I changed the annotation to use explicit Hibernate type to be in line with the testing goals. The generator related annotations are not mandatory here and for that reason I deleted them.